hack/bazel.sh

```diff
@@ -20,6 +20,20 @@ set -o errexit
 set -o pipefail
 
 code=0
+# Remove retries after
+# https://github.com/bazelbuild/bazel/issues/12599
+BAZEL_SYNC_PLEASE='true'
+if [ "${BAZEL_SYNC_PLEASE:-}" = 'true' ]; then
+  for i in {1..3}; do
+    if (set -o xtrace && bazel sync); then
```

Comment:
note: this fetches all deps. we could instead `bazel fetch <target pattern>` but then we'd need to supply just the target pattern and keep that in sync (flags to other commands are not compatible with this one)

Reply:
also: in test-infra our build/test target is `//...` (all targets) and this is generally true, so it's pretty unlikely to be worth the complexity of fetch instead of sync.

Reply:
Given that this is behind a flag anyway, why not `bazel fetch $BAZEL_SYNC_PLEASE` and change it to `BAZEL_SYNC_PLEASE=//...` instead of `=true`?

Reply:
Things like pushing images, deploying k8s etc can suffer these types of flakes too, will benefit from this retry pattern but are scoped to something smaller than `//...`

Reply:
> Given that this is behind a flag anyway, why not `bazel fetch $BAZEL_SYNC_PLEASE` and change it to BAZEL_SYNC_PLEASE=//... instead of =true?

This SGTM, I was initially concerned about:
- Two sources of truth for the targets 
  - We're going to fallback to ensuring whatever dependencies we need in the `bazel test` / `bazel build` call anyhow, so not a _big_ deal, just another place to try to keep this up to date
- Target patterns can be like `//... -//foo` (multiple args / an array). If bazel doesn't allow spaces in targets (seems likely actually?) then we can just space/word split the string.

Agree that this is desirable for smaller targets, my initial perusal found `//...` in use.
I'll verify patterns and no-spaces support and rework this.

Reply:
Does not appear that spaces are valid in targets so splitting should be fine.

https://docs.bazel.build/versions/master/build-ref.html#name

> Target names must be composed entirely of characters drawn from the set a–z, A–Z, 0–9, and the punctuation symbols !%-@^_` "#$&'()*-+,;<=>?[]{|}~/.

While target patterns appear to be a superset consisting ~only of adding the subtractive targets (preceding `-`), and the `...` wildcarding.

https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/runtime/commands/target-syntax.txt

Reply:
even better: we don't need to split at all. `bazel fetch '//... -//experiment/...'` turns out to be perfectly valid.

Reply:
done.